src/views/app/TheIntegrations/components/PagoPlux.vue

```diff
@@ -0,0 +1,194 @@
+<script setup lang="ts">
+import { reactive, computed, onMounted } from "vue";
+
+import {
+  pagopluxClientIdRules,
+  pagopluxSecretKeyRules,
+  rucRules
+} from '@/utils/validations'
+
+import pagopluxImage from "@/assets/integrations/pagoplux.png";
+import usePaymentMethodsStore from '@/store/paymentMethods';
+import ToggleInput from "@/components/ToggleInput.vue";
+import useClientStore from '@/store/client'
+import Card from "@/components/Card.vue";
+import useAuthStore from '@/store/auth';
+
+const authStore = useAuthStore();
+const clientStore = useClientStore();
+const paymentMethodsStore = usePaymentMethodsStore();
+
+const pagopluxForm = reactive({
+  ruc: "",
+  clientId: "",
+  secretKey: "",
+  active: false,
+  formVisible: false,
+});
+
+const toggleText = computed(() => {
+  return pagopluxForm.active ? "Desactivar" : "Activar";
+});
+
+const rules = {
+  ruc: rucRules,
+  clientId: pagopluxClientIdRules,
+  secretKey: pagopluxSecretKeyRules
+};
+
+const isFormValid = computed(() => {
+  const isRucValid = rucRules.every(rule => rule.validate(pagopluxForm.ruc));
+  const isClientIdValid = pagopluxClientIdRules.every(rule => rule.validate(pagopluxForm.clientId));
+  const isSecretKeyValid = pagopluxSecretKeyRules.every(rule => rule.validate(pagopluxForm.secretKey));
+  return isRucValid && isClientIdValid && isSecretKeyValid;
+});
+
+function isActive(event: boolean): void {
+  pagopluxForm.active = event;
+  if(!event) {
+    pagopluxForm.ruc = '';
+    pagopluxForm.clientId = '',
+    pagopluxForm.secretKey = ''
+  };
+  pagopluxForm.formVisible = event;
+};
+
+async function sendPagoPluxData () {
+  const data = {
+    restaurantRUC: pagopluxForm.ruc,
+    clientToken: `${pagopluxForm.clientId}:${pagopluxForm.secretKey}`
+  };
+  const restaurantUuid = clientStore.client?.restaurant?.uuid;
+  await paymentMethodsStore.putPagopluxData(data, restaurantUuid!);
+  pagopluxForm.formVisible = false;
+}
+
+onMounted( async () => {
+  const userAuth = await authStore.checkAuth();
+
+  await clientStore.getClientByEmail(userAuth?.email!);
+
+  await paymentMethodsStore.getPaymentMethods(clientStore.client?.restaurant?.uuid!)
+})
+</script>
+
+<template>
+  <p class="title">
+    <li>
+      Configura tu cuenta de pagoplux 😊
+    </li>
+  </p>
+  <Card :class="{ expanded: pagopluxForm.active }">
+    <template #title>
+      <div class="form-integration">
+        <p class="form-integration-title">Pagoplux</p>
+        <img :src="pagopluxImage" alt="logo de pagoplux" class="form-integration-image" />
+      </div>
+    </template>
+    <template #content>
+      <form class="form">
+        <ToggleInput
+          v-model:value="pagopluxForm.active"
+          :text="toggleText"
+          @update:modelValue="isActive" />
+        <div class="form-content" v-show="pagopluxForm.active && pagopluxForm.formVisible">
```

Comment:
this can be just a computed property instead making the comparison on the conditional

Reply:
oki